xapian-letor/ranker/svmranker.cc

```diff
@@ -45,6 +45,20 @@ Proceedings of the eighth ACM SIGKDD international conference on Knowledge disco
 using namespace std;
 using namespace Xapian;
 
+static void
+clear_svm_problem(svm_problem problem, int n)
+{
+    delete [] problem.y;
+    problem.y = NULL;
+    for (int i = 0; i < n;i++)
+    {
```

Comment:
probably best to run xapian-check-patch before committing changes. 
I have also been trying to form a habit of running Valgrind before pushing a new feature. That helps in not introducing a memory leak.